CIP-XXXX/README.md

```diff
@@ -0,0 +1,230 @@
+
+#### Scope
+
+Particular endpoint specs for the query layer are TBD.
+
+Here's a proposed list of endpoints that are needed:
+
+##### UTxO queries
+
+Get UTxOs by address - adapt [kupo-like interface](https://cardanosolutions.github.io/kupo/#section/Patterns) that allows to query by address components with wildcards.
+
+##### Staking and rewards
+
+- Get staking rewards by stake address
+- Get staking history by stake address
+
+##### Era summaries
+
+- Get a summary of eras
+
+##### Epochs & protocol parameters
+
+- Get latest epoch's protocol parameters
+- Get lastest epoch info
+- Get any epoch's protocol parameters (by epoch number)
+- Get any epoch's info (by epoch number)
+
+##### Data by hash
+
+- Get scripts by hash
+- Get datums by hash
+
+##### Transactions
+
+- Transaction contents by hash
+- Transaction status by hash
+
+##### Assets
+
+- Asset history by policy ID (mint/burn)
+
+##### Stake pools
+
+- Get list of stake pools
+- Get pool parameters by pool ID
+- Get pool history by pool ID
+- Get pool metadata by pool ID
+- Get pool relays by pool ID
+- Get pool delegators by pool ID
+- Get pool blocks by pool ID
+- Get pool certificates by pool ID
+
+### Constraints interface
+
+Constraints interface is a way for dApps to create transactions without having to deal with transaction structure. Effectively, when constraints interface is used, the wallet becomes fully responsible for tasks such as balancing, change generation, fee estimation and CBOR encoding.
+
+While freeing dApp developers from having to deal with low-level details, constraints interface opens a possibility to implement wallet-specific features that would be impossible to have with CIP-30, using which dApp assumes full control. For example:
+
+- ability to edit a transaction to perform additional actions: for example, sending ADA or tokens
+- merging multiple transactions from different dApps together (assuming script validators are written in a way that allows that)
+- ensuring change distribution between addresses: for example, multi-delegation feature that
+
+#### Scope
+
+A TBD list of constraints for dApps to use.
+
+- Pay to address (with optional datum and optional reference script)
+- Spend a UTxO
+  * PubKey address
+  * Native Script address
+  * Plutus Script address
+  * Native Script address via a reference input
+  * Plutus Script address via a reference input
+- Reference a UTxO
+- Mint currency
+  * Plutus Scripts
+  * Native Scripts
+  * Plutus Scripts via reference inputs
+  * Native Scripts via reference inputs
+- Set validity interval
+- Require signature from PubKey
+- Set transaction validity to `false`
+- Delegate stake to
+  * Plutus Script
+  * Native Script
+  * PubKey
+- Withdraw stake from
+  * Plutus Script
+  * Native Script
+  * PubKey
+- Register stake
+  * ScriptHash
+  * PubKey
+- Deregister stake scripts / pubkeys
+  * Native Script
+  * Plutus Script
+  * PubKey
+
+### JsonTx API
```

Comment:
It would be great to specify this as a separate CIP, so that it could be maintained independently of dapp connector and other CIPs could reference it. @KtorZ seems to have done some work on this already https://github.com/CardanoSolutions/cardanonical